### PR TITLE
Add an option to use icon-only buttons on grids

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Branch?         | 1.13
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Before
Capture d’écran 2022-03-22 à 11 20 58

After
Capture d’écran 2022-03-22 à 11 22 20

### CODE REVIEW

**Comment by loic425** [comment]
After this one, I will create a PR on grid documentation.

**Comment by loic425** [comment]
@pamil @lchrusciel This one is easy to review and allow to use the already existing labeled options of the `@SyliusUi/Macro/buttons.html.twig` macros.

```yaml
sylius_grid:
    grids:
        my_grid:
            actions:
                main:
                    create:
                        type: create
                        labeled: false
                item:
                    update:
                        type: update
                        options:
                            labeled: false
                    delete:
                        type: delete
                        options:
                            labeled: false
```

**Review by vvasiloi — APPROVED** [review]
Don't forget the documentation! 😉

**Comment by loic425** [comment]
> Don't forget the documentation! 😉

https://github.com/Sylius/Sylius/pull/13440/files

File: src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/applyTransition.html.twig [hunk 1]
```diff
@@ -1,9 +1,12 @@
+{% set labeled = options.labeled is defined ? options.labeled : true %}
+
 {% if sm_can(data, options.transition, options.graph) %}
     <form action="{{ path(options.link.route, options.link.parameters) }}" method="post">
         <input type="hidden" name="_csrf_token" value="{{ csrf_token(data.id) }}">
         <input type="hidden" name="_method" value="PUT">
-        <button class="ui loadable {{ options.class|default }} labeled icon button" type="submit" data-js-disable=".sylius-grid-table-wrapper button, .sylius-grid-table-wrapper a">
-            <i class="{{ action.icon }} icon"></i> {{ action.label|trans }}
+        <button class="ui loadable {{ options.class|default }} {% if labeled %}labeled{% endif%} icon button" type="submit" data-js-disable=".sylius-grid-table-wrapper button, .sylius-grid-table-wrapper a">
```
Comment (lchrusciel) [inline src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/applyTransition.html.twig:7]:
```suggestion
        <button class="ui loadable {{ options.class|default }} {% if labeled %}labeled{% endif %} icon button" type="submit" data-js-disable=".sylius-grid-table-wrapper button, .sylius-grid-table-wrapper a">
```

**Review by lchrusciel — COMMENTED** [review]
Hey @loic425, 

can you revive this PR? Also, can you add some screenshots to the description? Before and after would be best :)

**Comment by loic425** [comment]
> Hey @loic425,
> 
> can you revive this PR? Also, can you add some screenshots to the description? Before and after would be best :)

done

File: src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/create.html.twig [hunk 2]
```diff
@@ -2,4 +2,5 @@
 
 {% set path = options.link.url|default(path(options.link.route|default(grid.requestConfiguration.getRouteName('create')), options.link.parameters|default([]))) %}
 
-{{ buttons.create(path, action.label) }}
+{% set labeled = options.labeled is defined ? options.labeled : true %}
+{{ buttons.edit(path, action.label, null, labeled) }}
```
[the remark this answers is not in the document]
Comment (loic425) [reply src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/create.html.twig:6]: I think I remember why, cause if we don’t define any label, the label is generated with the key.

```yaml
sylius_grid:
    grids:
        app_admin_supplier:
            # ...
            actions:
                main:
                    create:
                        type: create
                item:
                    update:
                        type: update
                    delete:
                        type: delete
```

Labels will be Create, Update and Delete.
With Grid builder, that’s the same behaviour.

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/create.html.twig (hunk 2 above)
[the remark this answers is not in the document]
Comment (loic425) [reply src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/create.html.twig:6]: labeled seems to be correct (US), labelled too (EN)

But Sylius twig macro already uses "labeled" as an argument
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/UiBundle/Resources/views/Macro/buttons.html.twig

And semantic ui uses "labeled" too
https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/UiBundle/Resources/views/Macro/buttons.html.twig#L2

[the remark this answers is not in the document]
Comment (loic425) [reply src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/create.html.twig:6]: with `edit` macro, you can see why I choose "options" for that.

[the remark this answers is not in the document]
Comment (lchrusciel) [reply src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/create.html.twig:6]: We need to fix this, I assume:
```suggestion
{{ buttons.create(path, action.label, null, labeled) }}
```
But I buy your explanation

[the remark this answers is not in the document]
Comment (jakubtobiasz) [reply src/Sylius/Bundle/UiBundle/Resources/views/Grid/Action/create.html.twig]: I fixed the thing Łukasz mentioned. I'm resolving as I think everything is clear c:.

**Review by jakubtobiasz — APPROVED** [review]

**Comment by jakubtobiasz** [comment]
Thank you, @loic425!

**Pull request merged**